src/Auth/Authentication/Actions/Email2FA.php

```diff
@@ -0,0 +1,144 @@
+<?php
```

Comment:
I don't think there's a reason to port this from Shield. It just makes a maintenance issue to keep it up to date to changes from Shield. Was there a valid reason to bring it forward?

Reply:
The reason to port this is that i include Themeable trait and made methods to use Auth config items to render templates

Reply:
Ah, gotcha. It looks like Shield is using the `settings()` command instead of `config`. 

I am really uncomfortable duplicating all of the code here. I think a better solution is to submit a PR to Shield for the actions to use a `renderView()` method in the controllers so that all that needs to be extended is that one method and all of the other logic stays the same. 

Please remove those changes from this PR and we can get the rest merged, and then work on the other changes.

Reply:
Thanks for updating that. Will merge.